wagtail_localize/strings.py

```diff
@@ -389,3 +389,28 @@ def restore_strings(template, strings):
         text_element.replaceWith(string.render_soup(attrs))
 
     return str(soup)
+
+
+
+def extract_ids(template):
+    """Extract link ids from one template string and return it in a set."""
+    soup = BeautifulSoup(template, "html.parser")
+    ids = set()
+    for element in soup.descendants:
+        if not isinstance(element, Tag):
+            continue
+
+        if element.name == 'a':
```

Comment:
I think we could get rid of this check, technically any inline tag with attributes will be given an `id` (like custom inline tags for example)

Reply:
Do you want me to remove that ?

I've made it because currently there is

```
        # Elements can't have attributes, except for <a> tags
        keys = set(element.attrs.keys())
        if element.name == 'a' and 'id' in keys:
            keys.remove('id')
        if keys:
            raise ValueError(_("Strings cannot have any HTML tags with attributes (except for 'id' in <a> tags)"))
```

Reply:
Oh, sorry I forgot I added that! Yep, this is fine how it is, we can deal with custom tags later